### DISCUSSION TITLE
Add output data after the simulation is over

### DISCUSSION THREAD

**Discussion by PengWei97** [idaholab/moose#17329] [Q&A General] [answered]
Dear MOOSE experts,
Recently, I recently completed a simulation and output the strain (the three strain components in 2D), but now I want to output the stress component again. Now, could I get the stress data on the basis of previous simulations without recalculating?
This is the moose example I used，[poly_grain_growth_2D_eldrforce.i](https://github.com/idaholab/moose/blob/next/modules/combined/examples/phase_field-mechanics/poly_grain_growth_2D_eldrforce.i). 

Any suggestions or recommendations to fix these problems would be greatly appreciated.
Thank you
Wei Peng

**Comment by aeslaughter** [comment]
Give this a try:

```
[Outputs]
  [out]
    type = Exodus
    additional_execute_on = FINAL
  []
[]
```

**Reply by PengWei97** [reply]
Thank you very much. @aeslaughter 

According to your suggestion, I modified the Outputs module and added sub-blocks that require output stress in AuxKernel, Input file is as follows
[poly_grain_growth_2D_eldrforce_0.i ](https://github.com/PengWei97/monkey/blob/main/problem/AddOutputData_notRecalculating/poly_grain_growth_2D_eldrforce_0.i)

and then executed the code, 
`mpirun -np 10 ~/projects/monkey/monkey-opt -i poly_grain_growth_2D_eldrforce_0.i > poly_grain_growth_2D_eldrforce_0.log `
But this method shows that it is recalculated. 

Could you talk more about it?

**Reply by aeslaughter** [reply]
Can you give me more information, I don't understand "But this method shows that it is recalculated." What method shows what is recalculated?

**Reply by Traiwit** [reply]
@aeslaughter I think he means, he wants to use stress calculated from his first simulation directly on his second simulation without re-calculating it. 

> Now, could I get the stress data on the basis of previous simulations without recalculating?

**Reply by PengWei97** [reply]
Yes, what I want to do is to directly obtain the stress component data on the basis of my first simulation, instead of recalculating the model after modifying the input file.
In detail, the exodus file now contains strain components (three in two dimensions) and the Euler angle of each grain. Knowing the elastic stiffness tensor, how to pass Hooke's law
to get the stress component?

**Reply by Traiwit** [reply]
Hi, for uploading the data from .e file, you might want to have a look at 'initial conditions + restart'
```
`use_for_exodus_restart = true

`initial_from_file_var`
```

but for the 2nd part of your q, I'm not sure either, I'm also very new to MOOSE.
Is it possible to calculate material property before running the FEM simulation? It might got something to do with AuxKernals

Could someone help with this? thank you :)

**Comment by hugary1995** [comment, marked as answer]
Wei, you should use solution aux [https://mooseframework.inl.gov/source/auxkernels/SolutionAux.html](https://mooseframework.inl.gov/source/auxkernels/SolutionAux.html) and solution user object [https://mooseframework.inl.gov/source/userobject/SolutionUserObject.html](https://mooseframework.inl.gov/source/userobject/SolutionUserObject.html) to read in your previous results.

In your Problem block, set solve = false, i.e.
```
[Problem]
  solve = false
[]
```
to skip the solve.

Keep everything else the same, and just add a stress calculator.

**Reply by aeslaughter** [reply]
This should work, but only disable the solve if the calculations being done in the second input file are not solving a PDE. Sorry I did understand your initial condition and lead you down the wrong path.

**Reply by PengWei97** [reply]
Thank you very much for your enthusiastic help. 
I am not very familiar with some functions of moose, so I solved this problem with paraview. Given the strain component, Euler angle, and elastic modulus without rotation, use the generalized Hooke's law to solve. Of course, if you has a more detailed plan based on moose, and I am happy to try it.
